includes/rest/class-outbox-controller.php

```diff
@@ -232,11 +232,15 @@ public function get_items( $request ) {
 	 * @param \WP_REST_Request $request Request object.
 	 * @return array Response object on success, or WP_Error object on failure.
 	 */
-	public function prepare_item_for_response( $item, $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		$type        = \get_post_meta( $item->ID, '_activitypub_activity_type', true );
-		$transformer = Factory::get_transformer( $item->post_content );
-		$activity    = $transformer->to_activity( $type );
+	public function prepare_item_for_response( $item, $request ) {
+		$type = \get_post_meta( $item->ID, '_activitypub_activity_type', true );
+
+		$activity = new Activity();
+		$activity->set_type( $type );
 		$activity->set_id( $item->guid );
+		// Pre-fill the Activity with data (for example cc and to).
+		$activity->set_object( \json_decode( $item->post_content, true ) );
+		$activity->set_actor( Actors::get_by_various( $request->get_param( 'user_id' ) )->get_id() );
```

Comment:
Looks like we don't need set_actor? It doesn't change the content of the actor property.
Is there a scenario where we grab posts that are not by the passed user and need to update the actor to be the passed user?

Reply:
On `Announces` the Activity-Actor is the Blog-Actor and the Object is attributed to the Author.